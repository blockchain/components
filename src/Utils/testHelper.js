import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
