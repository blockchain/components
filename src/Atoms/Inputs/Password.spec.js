import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Password from './Password'

describe('Password', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Password />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with type', () => {
    const tree = mountWithTheme(<Password type="password" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with error="error"', () => {
    const tree = mountWithTheme(<Password error="error" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with label="label"', () => {
    const tree = mountWithTheme(<Password label="label" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with label="label" and disabled', () => {
    const tree = mountWithTheme(<Password disabled label="label" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with label="label" and error="error"', () => {
    const tree = mountWithTheme(<Password error="error" label="label" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<Password width="200px" />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with disabled', () => {
    const tree = mountWithTheme(<Password disabled />)

    expect(tree.find(Password)).toMatchSnapshot()
  })

  it('should render correctly with switching its type', () => {
    const tree = mountWithTheme(<Password />)

    tree.find('svg').simulate('click')

    expect(tree.find(Password)).toMatchSnapshot()

    tree.find('svg').simulate('click')

    expect(tree.find(Password)).toMatchSnapshot()
  })
})
