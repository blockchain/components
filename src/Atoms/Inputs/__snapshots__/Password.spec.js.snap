// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password should render correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.input
            disabled={false}
            e2e="defaultInput"
            type="password"
            width="100%"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={false}
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c3"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with disabled 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={true}
  e2e="defaultInput"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={true}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={true}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.input
            disabled={true}
            e2e="defaultInput"
            type="password"
            width="100%"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={true}
              type="password"
              width="100%"
            />
          </styled.input>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with error="error" 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c3 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #F36331;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c3:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c3:active:not(:disabled),
.c3:focus:not(:disabled) {
  box-shadow: 0 0 0 1px transparent;
}

.c3::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.875rem;
  padding-bottom: 0.5rem;
  color: #F36331;
}

.c4 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  error="error"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    error="error"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      error="error"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.label
            disabled={false}
            error={true}
          >
            <label
              className="c2"
              disabled={false}
            >
              error
            </label>
          </styled.label>
          <styled.input
            disabled={false}
            e2e="defaultInput"
            error="error"
            type="password"
            width="100%"
          >
            <input
              className="c3"
              data-e2e="defaultInput"
              disabled={false}
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c4"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with label="label" 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c3 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c3:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c3:active:not(:disabled),
.c3:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c3::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.875rem;
  padding-bottom: 0.5rem;
  color: #757679;
}

.c4 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  label="label"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    label="label"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      label="label"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.label
            disabled={false}
            error={false}
          >
            <label
              className="c2"
              disabled={false}
            >
              label
            </label>
          </styled.label>
          <styled.input
            disabled={false}
            e2e="defaultInput"
            label="label"
            type="password"
            width="100%"
          >
            <input
              className="c3"
              data-e2e="defaultInput"
              disabled={false}
              label="label"
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c4"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with label="label" and disabled 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c3 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c3:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c3:active:not(:disabled),
.c3:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c3::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.875rem;
  padding-bottom: 0.5rem;
  color: #757679;
  cursor: not-allowed;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={true}
  e2e="defaultInput"
  label="label"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={true}
    e2e="defaultInput"
    label="label"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={true}
      e2e="defaultInput"
      label="label"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.label
            disabled={true}
            error={false}
          >
            <label
              className="c2"
              disabled={true}
            >
              label
            </label>
          </styled.label>
          <styled.input
            disabled={true}
            e2e="defaultInput"
            label="label"
            type="password"
            width="100%"
          >
            <input
              className="c3"
              data-e2e="defaultInput"
              disabled={true}
              label="label"
              type="password"
              width="100%"
            />
          </styled.input>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with label="label" and error="error" 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c3 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #F36331;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c3:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c3:active:not(:disabled),
.c3:focus:not(:disabled) {
  box-shadow: 0 0 0 1px transparent;
}

.c3::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.875rem;
  padding-bottom: 0.5rem;
  color: #F36331;
}

.c4 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  error="error"
  label="label"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    error="error"
    label="label"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      error="error"
      label="label"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.label
            disabled={false}
            error={true}
          >
            <label
              className="c2"
              disabled={false}
            >
              label - error
            </label>
          </styled.label>
          <styled.input
            disabled={false}
            e2e="defaultInput"
            error="error"
            label="label"
            type="password"
            width="100%"
          >
            <input
              className="c3"
              data-e2e="defaultInput"
              disabled={false}
              label="label"
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c4"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with switching its type 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="text"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="text"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.input
            disabled={false}
            e2e="defaultInput"
            type="text"
            width="100%"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={false}
              type="text"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c3"
              onClick={[Function]}
            >
              Hide
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with switching its type 2`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.input
            disabled={false}
            e2e="defaultInput"
            type="password"
            width="100%"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={false}
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c3"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with type 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  type="password"
  width="100%"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="password"
    width="100%"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="password"
      width="100%"
    >
      <styled.div
        className="c0"
        width="100%"
      >
        <div
          className="c0 c1"
          width="100%"
        >
          <styled.input
            disabled={false}
            e2e="defaultInput"
            type="password"
            width="100%"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={false}
              type="password"
              width="100%"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c3"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;

exports[`Password should render correctly with width 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 200px;
  position: relative;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 200px;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: box-shadow 0.25s ease-out;
  transition: box-shadow 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c3 {
  position: absolute;
  right: 1.25rem;
  bottom: 0.875rem;
  cursor: pointer;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.625rem;
  font-weight: 600;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 input {
  padding-right: 3.5rem;
}

<Password
  disabled={false}
  e2e="defaultInput"
  type="password"
  width="200px"
>
  <Styled(Input)
    disabled={false}
    e2e="defaultInput"
    renderSuffix={[Function]}
    type="password"
    width="200px"
  >
    <Input
      className="c0"
      disabled={false}
      e2e="defaultInput"
      renderSuffix={[Function]}
      type="password"
      width="200px"
    >
      <styled.div
        className="c0"
        width="200px"
      >
        <div
          className="c0 c1"
          width="200px"
        >
          <styled.input
            disabled={false}
            e2e="defaultInput"
            type="password"
            width="200px"
          >
            <input
              className="c2"
              data-e2e="defaultInput"
              disabled={false}
              type="password"
              width="200px"
            />
          </styled.input>
          <styled.span
            onClick={[Function]}
          >
            <span
              className="c3"
              onClick={[Function]}
            >
              Show
            </span>
          </styled.span>
        </div>
      </styled.div>
    </Input>
  </Styled(Input)>
</Password>
`;
